### PR TITLE
Handle interface methods in LF conversion

### PR DESCRIPTION
**Pull request by cocreature**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by sofiafaro-da — APPROVED** [review]
Great work, thank you!

File: compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs [hunk 1]
```diff
@@ -432,11 +432,13 @@ convertInterfaces env tyThings = interfaceClasses
         [ convertChoice arg res
         |
           TypeCon (NameIn DA_Internal_Template_Functions "HasExercise") [_, arg, res] <- classSCTheta cls]
+      -- Drop toIface/fromIface/toIfaceContractId/fromIfaceContractId to get only user-defined methods.
+      methods <- mapM convertMethod (drop 4 $ classMethods cls)
```
[the remark this answers is not in the document]
Comment (cocreature) [reply compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs:436]: yeah agree, I’ll add an issue.

[the remark this answers is not in the document]
Comment (sofiafaro-da) [reply compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs:436]: Thanks!  🙇‍♀️

**Comment by sofiafaro-da** [comment]
Part of #11006

File: compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (cocreature) [reply compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs:436]: https://github.com/digital-asset/daml/issues/11071

**Pull request merged**